### PR TITLE
[Feature] Add `crowdhuman` dataset

### PR DESCRIPTION
**Pull request by hhaAndroid**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation for this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit tests to ensure the correctness.
3. If the modification has a potential influence on downstream projects, this PR should be tested with downstream projects, like MMDetection or MMClassification.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: configs/yolov5/crowdhuman/yolov5_s-v61_fast_8xb16-300e_crowdhuman.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
+_base_ = '../yolov5_s-v61_syncbn_fast_8xb16-300e_coco.py'
+
+load_from = 'https://download.openmmlab.com/mmyolo/v0/yolov5/yolov5_s-v61_syncbn_fast_8xb16-300e_coco/yolov5_s-v61_syncbn_fast_8xb16-300e_coco_20220918_084700-86e02187.pth'  # noqa
```
Comment (Nioolek) [inline configs/yolov5/crowdhuman/yolov5_s-v61_fast_8xb16-300e_crowdhuman.py:3]: add doc: Use the model trained on the COCO as the pretrained model

File: configs/yolov5/crowdhuman/yolov5_s-v61_fast_8xb16-300e_crowdhuman.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+anchors = [
+    [(6, 14), (12, 28), (19, 48)],  # P3/8
+    [(29, 79), (46, 124), (142, 54)],  # P4/16
+    [(73, 198), (124, 330), (255, 504)]  # P5/32
+]
+
+model = dict(
+    bbox_head=dict(
+        head_module=dict(num_classes=num_classes),
+        prior_generator=dict(base_sizes=anchors)))
```
Comment (PeterH0323) [inline configs/yolov5/crowdhuman/yolov5_s-v61_fast_8xb16-300e_crowdhuman.py:21]: loss 权重也需要根据 number_class 重新赋值

File: configs/yolov5/README.md [hunk 3]
```diff
@@ -48,6 +48,17 @@ In the official YOLOv5 code, the `random_perspective` data augmentation in COCO
...
 
+### CrowdHuman
+
+| Backbone | size | SyncBN | AMP | Mem (GB) | box AP50(CrowDHuman Metric) |  MR  |  JI   |                                                             Config                                                              | Download |
+| :------: | :--: | :----: | :-: | :------: | :-------------------------: | :--: | :---: | :-----------------------------------------------------------------------------------------------------------------------------: | :------: |
+| YOLOv5-s | 640  |  Yes   | Yes |   2.6    |            85.79            | 48.7 | 75.33 | [config](https://github.com/open-mmlab/mmyolo/tree/master/configs/yolov5/crowdhuman/yolov5_s-v61_fast_8xb16-300e_crowdhuman.py) |          |
+
+**Note**:
+
+1. This is a very simple implementation that simply replaces COCO's anchor with the `tools/analysis_tools/optimize_anchors.py` script. We'll adjust the parameters later to improve performance.
```
Comment (Nioolek) [inline configs/yolov5/README.md:59]: `adjust the parameters` -> `adjust other parameters`

File: configs/yolov5/crowdhuman/yolov5_s-v61_fast_8xb16-300e_crowdhuman.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
+_base_ = '../yolov5_s-v61_syncbn_fast_8xb16-300e_coco.py'
+
+load_from = 'https://download.openmmlab.com/mmyolo/v0/yolov5/yolov5_s-v61_syncbn_fast_8xb16-300e_coco/yolov5_s-v61_syncbn_fast_8xb16-300e_coco_20220918_084700-86e02187.pth'  # noqa
+
+# dataset settings
+data_root = 'data/CrowdHuman/'
+dataset_type = 'YOLOv5CrowdHumanDataset'
+
+# parameters that often need to be modified
+num_classes = 1
```
Comment (PeterH0323) [inline configs/yolov5/crowdhuman/yolov5_s-v61_fast_8xb16-300e_crowdhuman.py:10]: metainfo 需要重写

File: configs/yolov5/crowdhuman/yolov5_s-v61_fast_8xb16-300e_crowdhuman.py (hunk 2 above)
Comment (hhaAndroid) [reply configs/yolov5/crowdhuman/yolov5_s-v61_fast_8xb16-300e_crowdhuman.py:21]: 就一个类，所以就懒得写了

File: configs/yolov5/crowdhuman/yolov5_s-v61_fast_8xb16-300e_crowdhuman.py (hunk 4 above)
Comment (hhaAndroid) [reply configs/yolov5/crowdhuman/yolov5_s-v61_fast_8xb16-300e_crowdhuman.py:10]: 不需要，因为我不是继承 cocodataset

File: configs/yolov5/crowdhuman/yolov5_s-v61_fast_8xb16-300e_crowdhuman.py [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+val_dataloader = dict(
+    dataset=dict(
+        type=dataset_type,
+        data_root=data_root,
+        ann_file='annotation_val.odgt',
+        data_prefix=dict(img='Images/'),
+        # CrowdHumanMetric does not support out-of-order output images
+        # for the time being. batch_shapes_cfg does not support.
+        batch_shapes_cfg=None))
```
Comment (Nioolek) [inline configs/yolov5/crowdhuman/yolov5_s-v61_fast_8xb16-300e_crowdhuman.py:38]: 这里没有使用batch_shapes_cfg，那为啥还要单独建一个 `YOLOv5CrowdHumanDataset` 类？

Comment (hhaAndroid) [reply configs/yolov5/crowdhuman/yolov5_s-v61_fast_8xb16-300e_crowdhuman.py:39]: 未来要支持的

Comment (hhaAndroid) [reply configs/yolov5/crowdhuman/yolov5_s-v61_fast_8xb16-300e_crowdhuman.py:39]: 这算是一个 bug 了

**Review by PeterH0323 — APPROVED** [review]
LGTM

**Pull request merged**